botorch/optim/initializers.py

```diff
@@ -63,6 +63,93 @@
 ]
 
 
+def transform_constraints(
+    constraints: Union[List[Tuple[Tensor, Tensor, float]], None], q: int, d: int
+) -> List[Tuple[Tensor, Tensor, float]]:
+    """Transform constraints to sample from a d*q dimensional space instead of a
+    d dimensional state.
+
+    Args:
+        constraints (Union[List[Tuple[Tensor, Tensor, float]], None]): Constraints
+            to transform.
+        q (int): Size of the q-batch.
+        d (int): Dimensionality of the problem.
+
+    Returns:
+        List[Tuple[Tensor, Tensor, float]]: List of transformed constraints.
+    """
+    if constraints is None:
+        return None
+    transformed = []
+    for constraint in constraints:
+        if len(constraint[0].shape) == 1:
+            transformed += transform_intra_point_constraint(constraint, d, q)
+        else:
+            transformed.append(transform_inter_point_constraint(constraint, d))
+    return transformed
+
+
+def transform_intra_point_constraint(
+    constraint: Tuple[Tensor, Tensor, float], d: int, q: int
+) -> List[Tuple[Tensor, Tensor, float]]:
+    """Transforms an intra-point/pointwise constraint from
+    d-dimensional space to a d*q dimesional space.
+
+    Args:
+        constraint (Tuple[Tensor, Tensor, float]): Constraints
+            to transform.
+        d (int): Size of the q-batch.
+        q (int): Dimensionality of the problem.
+
+    Raises:
+        ValueError: If indices in the constraints are larger than the
+            dimensionality d of the problem.
+
+    Returns:
+        List[Tuple[Tensor, Tensor, float]]: List of transformed constraints.
+    """
+    indices, coefficients, rhs = constraint
+    if indices.max() >= d:
+        raise ValueError("d has to be larger than the largest index in the constraint.")
```

Comment:
Kind of a weird way to put this, how about
```suggestion
        raise ValueError(f"Constraint indices cannot exceed the problem dimension {d=}.")
```